src/mlpack/core/util/size_checks.hpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file size_checks.hpp
+ * @author Kirill Mishchenko
+ * @author Bisakh Mondal
+ *
+ * Utility for checking same size & same dimensionality.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_UTIL_SIZE_CHECKS_HPP
+#define MLPACK_UTIL_SIZE_CHECKS_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace util {
+
+/**
+  * Check for if the given data points & labels have same size.
+  *
+  * @param data data.
+  * @param labels Labels.
+  * @param callerDescription A description of the caller that can be used for
+  *     error generation.
+  * @param mode For nature of comparision(default "CE").
+  *   types of mode:
+  *     "CE" equivalent to (data.n_cols, labels.n_elem).
+  *     "CC" equivalent to (data.n_cols, labels.n_cols).
+  * @param addInfo  An additional information about labels that can be used for
+  *	precise error generation. Default is "labels". Another e.g. weights
+  */
+template<typename DataType, typename LabelsType>
+inline void CheckSameSizes(const DataType& data,
+                           const LabelsType& labels,
+                           const std::string& callerDescription,
+                           const std::string& mode = "CE",
+                           const std::string& addInfo = "labels")
+{
+  if (mode != "CE" && mode != "CC")
+    // For development purpose, not intended for user.
+    Log::Fatal << "Ensure Providing Correct mode." << std::endl;
```

Comment:
Personally I would remove the mode if there is no use for it; some lines less we have to maintain on the other side a user might be confused why there is an option that will produce an error if used. If there is use for the mode in the future we can add it without breaking backward compatibility.

Reply:
Sure. I'll push the updates soon. Thanks for the review.

Reply:
I would agree with @zoq's input here. :+1: